controllers/numaresourcesoperator_controller_test.go

```diff
@@ -138,224 +141,242 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 				{MatchLabels: label1},
 				{MatchLabels: label2},
 			})
+			nroKey = client.ObjectKeyFromObject(nro)
 
 			mcp1 = testutils.NewMachineConfigPool("test1", label1, &metav1.LabelSelector{MatchLabels: label1}, &metav1.LabelSelector{MatchLabels: label1})
 			mcp2 = testutils.NewMachineConfigPool("test2", label2, &metav1.LabelSelector{MatchLabels: label2}, &metav1.LabelSelector{MatchLabels: label2})
 		})
 
-		Context("on the first iteration", func() {
-			It("should create CRD, machine configs and wait for MCPs updates", func() {
-				reconciler, err := NewFakeNUMAResourcesOperatorReconciler(platform.OpenShift, nro, mcp1, mcp2)
-				Expect(err).ToNot(HaveOccurred())
+		Context("with machine config pool with SIMPLE machine config selector", func() {
 
-				key := client.ObjectKeyFromObject(nro)
-				result, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
-				Expect(err).ToNot(HaveOccurred())
-				Expect(result).To(Equal(reconcile.Result{RequeueAfter: time.Minute}))
+			BeforeEach(func() {
+				var err error
 
-				crd := &apiextensionsv1.CustomResourceDefinition{}
-				key = client.ObjectKey{
-					Name: "noderesourcetopologies.topology.node.k8s.io",
-				}
-				Expect(reconciler.Client.Get(context.TODO(), key, crd)).ToNot(HaveOccurred())
+				reconciler, err = NewFakeNUMAResourcesOperatorReconciler(platform.OpenShift, nro, mcp1, mcp2)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			Context("on the first iteration", func() {
+				var firstLoopResult reconcile.Result
+				BeforeEach(func() {
+					var err error
 
-				mc := &machineconfigv1.MachineConfig{}
+					// call first reconcile loop
```

Comment:
this is already pretty evident from the code, though :)

Reply:
yep, I will delete that comment.